src/Examine/LuceneEngine/Directories/IDirectoryFactory.cs

```diff
@@ -8,5 +9,9 @@ public interface IDirectoryFactory
     {
         [SecuritySafeCritical]
         Directory CreateDirectory(LuceneIndexer indexer, string luceneIndexFolder);
+
+        bool IsReadOnly { get; }
+        [SecuritySafeCritical]
+        MergePolicy GetMergePolicy(IndexWriter writer);
```

Comment:
This is a breaking change so will need to work around this differently. There's a few options: 

* Create another interface like IDirectoryFactory2 and then an ext method to do the type checking and then use that ext method throughout the codebase
* Don't attach this method to IDirectoryFactory - it's not really much to do with the Directory after all and instead there could be a new event on LuceneIndexer like CreatingIndexWriter. This is probably the better option and also provides devs with a lot more flexibility too since they can then configure the writer even further if necessary.

Reply:
@Shazwazza I will look into that later as I made some additional changes later in v1, which I would need backport there :)

Reply:
I backport solution which works perfectly for me on v1, I think after refactor it shouldnt contain any breaking changes, but can you confirm? :)